ivadomed/loader/utils.py

```diff
@@ -646,25 +647,25 @@ def create_bids_dataframe(self):
             ext_ct = ('.nii.gz', '.nii')
             suffix_ct = ('ct', 'CT')
             force_index = []
-            for root, dirs, files in os.walk(path_data):
-                for file in files:
+            for path_object in path_data.glob('**/*'):
+                if path_object.is_file():
                     # Microscopy
-                    if file == "samples.tsv" or file == "samples.json":
-                        force_index.append(file)
-                    if (file.endswith(ext_microscopy) and os.path.basename(root) == "microscopy" and
-                            (root.replace(path_data, '').startswith("sub"))):
-                        force_index.append(os.path.join(root.replace(path_data, '')))
+                    if path_object.name == "samples.tsv" or path_object.name == "samples.json":
+                        force_index.append(path_object.name)
+                    if (path_object.name.endswith(ext_microscopy) and path_object.parent.name == "microscopy" and
+                            (str(path_object.parent).replace(str(path_data) + os.sep, '').startswith("sub"))):
+                        force_index.append(str(path_object.parent).replace(str(path_data) + os.sep, ''))
                     # CT-scan
-                    if (file.endswith(ext_ct) and file.split('.')[0].endswith(suffix_ct) and
-                            (os.path.basename(root) == "anat" or os.path.basename(root) == "ct") and
-                            (root.replace(path_data, '').startswith("sub"))):
-                        force_index.append(os.path.join(root.replace(path_data, '')))
+                    if (path_object.name.endswith(ext_ct) and path_object.name.split('.')[0].endswith(suffix_ct) and
+                            (path_object.parent.name == "anat" or path_object.parent.name == "ct") and
+                            (str(path_object.parent).replace(str(path_data) + os.sep, '').startswith("sub"))):
```

Comment:
Refactor this logic as before with regarding to simplify the `str.replace` portion. There shouldn't be the need to stringify path object just take a portion out SO that we can check for them starting with `sub` string.